File: next/security/user and permission/backend-ms1/src/main/java/ma/zs/stocky/service/impl/admin/security/ModelPermissionUserAdminServiceImpl.java
package ma.zs.stocky.service.impl.admin.security;


import ma.zs.stocky.bean.core.security.ModelPermissionUser;
import ma.zs.stocky.dao.criteria.core.security.ModelPermissionUserCriteria;
import ma.zs.stocky.dao.facade.core.security.ModelPermissionUserDao;
import ma.zs.stocky.dao.specification.core.security.ModelPermissionUserSpecification;
import ma.zs.stocky.service.facade.admin.security.ModelPermissionUserAdminService;
import ma.zs.stocky.zynerator.service.AbstractServiceImpl;
import ma.zs.stocky.zynerator.util.ListUtil;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.ArrayList;





import org.springframework.beans.factory.annotation.Autowired;

import ma.zs.stocky.service.facade.admin.security.UserAdminService ;
import ma.zs.stocky.bean.core.security.User ;
import ma.zs.stocky.service.facade.admin.security.ModelPermissionAdminService ;
import ma.zs.stocky.bean.core.security.ModelPermission ;
import ma.zs.stocky.service.facade.admin.security.ActionPermissionAdminService ;
import ma.zs.stocky.bean.core.security.ActionPermission ;

import java.util.List;
@Service
public class ModelPermissionUserAdminServiceImpl extends AbstractServiceImpl<ModelPermissionUser, ModelPermissionUserCriteria, ModelPermissionUserDao> implements ModelPermissionUserAdminService {







    public List<ModelPermissionUser> findByActionPermissionId(Long id){
        return dao.findByActionPermissionId(id);
    }
    public int deleteByActionPermissionId(Long id){
        return dao.deleteByActionPermissionId(id);
    }
    public long countByActionPermissionId(Long id){
        return dao.countByActionPermissionId(id);
    }
    public List<ModelPermissionUser> findByModelPermissionId(Long id){
        return dao.findByModelPermissionId(id);
    }
    public int deleteByModelPermissionId(Long id){
        return dao.deleteByModelPermissionId(id);
    }
    public long countByModelPermissionId(Long id){
        return dao.countByModelPermissionId(id);
    }
    public List<ModelPermissionUser> findByUserId(Long id){
        return dao.findByUserId(id);
    }
    public int deleteByUserId(Long id){
        return dao.deleteByUserId(id);
    }
    public long countByUserId(Long id){
        return dao.countByUserId(id);
    }






    public void configure() {
        super.configure(ModelPermissionUser.class, ModelPermissionUserSpecification.class);
    }


    @Autowired
    private UserAdminService userService ;
    @Autowired
    private ModelPermissionAdminService modelPermissionService ;
    @Autowired
    private ActionPermissionAdminService actionPermissionService ;

    public ModelPermissionUserAdminServiceImpl(ModelPermissionUserDao dao) {
        super(dao);
    }

}

File: nest/security/backend-ms1/src/app/module/service/admin/security/ModelPermissionUserAdminServiceImpl.ts
import {Injectable, NotFoundException} from "@nestjs/common";

import {AbstractServiceImpl} from "src/app/zynerator/service/AbstractServiceImpl";

import {ModelPermissionUserDao} from "src/app/shared/dao/facade/core/security/ModelPermissionUserDao";
import {ModelPermissionUserCriteria} from "src/app/shared/dao/criteria/core/security/ModelPermissionUserCriteria";
import {ModelPermissionUser} from "src/app/shared/bean/core/security/ModelPermissionUser";
import {ModelPermissionUserDto} from "src/app/shared/dto/security/ModelPermissionUserDto";


@Injectable()
export class ModelPermissionUserAdminServiceImpl extends AbstractServiceImpl<ModelPermissionUser, ModelPermissionUserCriteria, ModelPermissionUserDao>{

    constructor(private readonly dao: ModelPermissionUserDao ,
    ) {
        super(dao);
    }

    async save(item: ModelPermissionUser): Promise<ModelPermissionUser> {
        const saved = await this.dao.save(item);
        return saved;
    }


    async update(item: ModelPermissionUser): Promise<ModelPermissionUser> {
        const saved = await this.dao.saveOrUpdate(item);
        return saved;
    }

    async updateMultiple(items: ModelPermissionUser[]): Promise<void> {
        if (items) {
            items.forEach(e => this.update(e))
        }
    }

    async  findAllOptimized(): Promise<ModelPermissionUserDto[]> {
        return this.dao.findAllOptimized();
    }

    async findAll(): Promise<ModelPermissionUser[]> {
        return this.dao.findAll();
    }

    async findById(id: number): Promise<ModelPermissionUser> {
        return this.dao.findById(id);
    }


    async deleteById(id: number): Promise<number> {
        const existing = await this.findById(id);
        if (!existing) {
            throw new NotFoundException(`Model permission user with ID ${id} not found.`);
        }
        const result = await this.dao.deleteById(existing.id);
        return result;
    }

    async deleteMultiple(items: ModelPermissionUser[]): Promise<ModelPermissionUser[]> {
        const deletedItems: ModelPermissionUser[] = [];
        for (const item of items) {
            await this.deleteById(item.id);
            deletedItems.push(item);
        }
        return deletedItems;
    }

    async findByActionPermissionId(id: number): Promise<ModelPermissionUser[]> {
        return this.dao.findByActionPermissionId(id);
    }

    async deleteByActionPermissionId(id: number): Promise<number> {
        return this.dao.deleteByActionPermissionId(id);
    }
    async findByModelPermissionId(id: number): Promise<ModelPermissionUser[]> {
        return this.dao.findByModelPermissionId(id);
    }

    async deleteByModelPermissionId(id: number): Promise<number> {
        return this.dao.deleteByModelPermissionId(id);
    }
    async findByUserId(id: number): Promise<ModelPermissionUser[]> {
        return this.dao.findByUserId(id);
    }

    async deleteByUserId(id: number): Promise<number> {
        return this.dao.deleteByUserId(id);
    }

    async findWithAssociatedLists(id: number): Promise<ModelPermissionUser> {
        const result = await this.dao.findById(id);
    return result;
    }

    async updateWithAssociatedLists(item: ModelPermissionUser): Promise<ModelPermissionUser> {
         return await this.dao.saveOrUpdate(item);
    }
}

